sycl/include/sycl/handler.hpp

```diff
@@ -1028,6 +1054,21 @@ class __SYCL_EXPORT handler {
                                 KernelType KernelFunc) {
```

Comment:
Haven't yet looked up the call stack, but in SYCL 2020 spec we have:

```c++
  // Parameter pack acts as-if: Reductions&&... reductions, const KernelType
  // &kernelFunc
  template <typename KernelName, int Dimensions, typename... Rest>
  void parallel_for(range<Dimensions> numWorkItems, Rest&&... rest); // (1)

   // Deprecated in SYCL 2020.
  template <typename KernelName, typename KernelType, int Dimensions>
  void parallel_for(range<Dimensions> numWorkItems,
                    id<Dimensions> workItemOffset,
                    const KernelType& kernelFunc); // (2)

  template <int Dimensions>
  void parallel_for(range<Dimensions> numWorkItems, const kernel& kernelObject); // (3)
```

I guess that we can kind of ignore (2) for now and just say that yes, it has bugs.
(1) I assume is what you are fixing.
(3) is reserved for interoperability with backends, so I guess users of this overload should be aware of any backend-specific limitations.

I don't think that any extra changes are needed, but I'm adding this comment to make sure that we haven't missed any cases and that we are aligned on how those cases should be handled